packages/core/src/utils.ts

```diff
@@ -116,10 +116,14 @@ export class Utils {
   ) {
     if (typeof cid === 'string') cid = CID.parse(cid.replace('ipfs://', ''))
     let format = await ipfsApi.codecs.getCodec(cid.code).then((f) => f.name)
-    // The try catch behaviour of the js-ipfs-http-client was causing timing issues (socket hangup)
-    // https://github.com/ipfs/js-ipfs/blob/master/packages/ipfs-http-client/src/block/put.js#L34
-    if (format === 'dag-cbor') format = 'cbor'
-    else if (format === 'dag-pb') format = 'protobuf'
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    if (!ipfsApi.libp2p) {
```

Comment:
this will help our unit tests pass, but won't actually allow Ceramic v2 to be used with js-ipfs using the http client.  Do we want to do anything to try to detect whether it's go-ipfs or js-ipfs on the other end of the http client?

Reply:
looks like the http-client has a `version` command that maybe would tell us what we need?

https://github.com/ipfs/js-ipfs/blob/master/packages/ipfs-http-client/src/version.js

Reply:
I don't think that's possible? Or is there a way to detect that?

I think it's fine if we only support go-ipfs though the http-client. Happy to be proven wrong though.

Reply:
`version` output on `go-ipfs`:
```
{
    "Version": "0.12.0",
    "Commit": "06191df",
    "Repo": "12",
    "System": "amd64/linux",
    "Golang": "go1.16.12"
}
```
`version` output on `js-ipfs`:
```
{
    "Version": "0.14.1",
    "Commit": "",
    "Repo": "12",
    "ipfs-core": "0.14.1",
    "interface-ipfs-core": "^0.154.1",
    "ipfs-http-client": "^56.0.1"
}
```

Reply:
Ok, would be great if we could pick this info out on startup and bubble it down to this util function somehow. It doesn't make sense to call it every time.

Reply:
`id` output on `go-ipfs`:
```
{
    "ID": "12D3KooWS98rUgj8dy22Mo5AkrAv1EWSBbFeAVGXh8GXViLFsKtG",
    "PublicKey": "CAESIPKJ+JAFJko/UnfdHJHcaqlZjxDvk35iMjugqaZR+4v1",
    "Addresses": [
        "/ip4/127.0.0.1/tcp/4001/p2p/12D3KooWS98rUgj8dy22Mo5AkrAv1EWSBbFeAVGXh8GXViLFsKtG",
        "/ip4/127.0.0.1/udp/4001/quic/p2p/12D3KooWS98rUgj8dy22Mo5AkrAv1EWSBbFeAVGXh8GXViLFsKtG",
        "/ip4/136.56.26.176/udp/50241/quic/p2p/12D3KooWS98rUgj8dy22Mo5AkrAv1EWSBbFeAVGXh8GXViLFsKtG",
        "/ip4/172.18.0.2/tcp/4001/p2p/12D3KooWS98rUgj8dy22Mo5AkrAv1EWSBbFeAVGXh8GXViLFsKtG",
        "/ip4/172.18.0.2/udp/4001/quic/p2p/12D3KooWS98rUgj8dy22Mo5AkrAv1EWSBbFeAVGXh8GXViLFsKtG"
    ],
    "AgentVersion": "go-ipfs/0.12.0/06191df",
    "ProtocolVersion": "ipfs/0.1.0",
    "Protocols": [
        "/floodsub/1.0.0",
        "/ipfs/bitswap",
        "/ipfs/bitswap/1.0.0",
        "/ipfs/bitswap/1.1.0",
        "/ipfs/bitswap/1.2.0",
        "/ipfs/id/1.0.0",
        "/ipfs/id/push/1.0.0",
        "/ipfs/lan/kad/1.0.0",
        "/ipfs/ping/1.0.0",
        "/libp2p/autonat/1.0.0",
        "/libp2p/circuit/relay/0.1.0",
        "/libp2p/circuit/relay/0.2.0/stop",
        "/meshsub/1.0.0",
        "/meshsub/1.1.0",
        "/p2p/id/delta/1.0.0",
        "/x/"
    ]
}
```
`id` output on `js-ipfs`:
```
{
    "ID": "12D3KooWD54oQGUJnYPPCf7bKhRfJUsxjRR8KwMw8AEj2K8MgwuG",
    "PublicKey": "CAESIDBXIofLdANwZR5WhAM5axrFhf00hSHa68rsiG1RMR0n",
    "Addresses": [
        "/ip4/127.0.0.1/tcp/4002/p2p/12D3KooWD54oQGUJnYPPCf7bKhRfJUsxjRR8KwMw8AEj2K8MgwuG",
        "/ip4/127.0.0.1/tcp/4003/ws/p2p/12D3KooWD54oQGUJnYPPCf7bKhRfJUsxjRR8KwMw8AEj2K8MgwuG",
        "/ip4/172.18.0.2/tcp/4002/p2p/12D3KooWD54oQGUJnYPPCf7bKhRfJUsxjRR8KwMw8AEj2K8MgwuG",
        "/ip4/172.18.0.2/tcp/4003/ws/p2p/12D3KooWD54oQGUJnYPPCf7bKhRfJUsxjRR8KwMw8AEj2K8MgwuG"
    ],
    "AgentVersion": "js-ipfs/0.14.1",
    "ProtocolVersion": "ipfs/0.1.0",
    "Protocols": [
        "/floodsub/1.0.0",
        "/ipfs/bitswap/1.0.0",
        "/ipfs/bitswap/1.1.0",
        "/ipfs/bitswap/1.2.0",
        "/ipfs/id/1.0.0",
        "/ipfs/id/push/1.0.0",
        "/ipfs/lan/kad/1.0.0",
        "/ipfs/ping/1.0.0",
        "/libp2p/circuit/relay/0.1.0",
        "/libp2p/fetch/0.0.1",
        "/meshsub/1.0.0",
        "/meshsub/1.1.0"
    ]
}
```

Reply:
The `AgentVersion` field in the `id` output seems to be the most relevant:

`AgentVersion": "go-ipfs/0.12.0/06191df`

`AgentVersion": "js-ipfs/0.14.1`

Reply:
@smrz2001 seems like we should have three different configurations available for `id`:
* `go-ipfs` + `js-ipfs-http-client`
* `js-ipfs` + `js-ipfs-http-client`
* `js-ipfs` (in proc)

Reply:
Do both js-ipfs ones give AgentVersion js-ipfs?

Reply:
> Ok, would be great if we could pick this info out on startup and bubble it down to this util function somehow. It doesn't make sense to call it every time.

Yes, makes sense. In fact, we the `id` command is already run indirectly when we call `isOnline` on the IPFS HTTP client for the healthcheck at startup.

We could replace that with a direct call to `id` where we collect the full response, use it to figure out whether IPFS is online, and also to determine the type of IPFS node.

Reply:
If you can, please create a PR with this change on top of this branch (and an issue in linear). Thanks!

Reply:
> Do both js-ipfs ones give AgentVersion js-ipfs?

Good, question, @oed, I'm assuming yes but will confirm. Will test and update the PR by Monday.